pydantic/_internal/_generate_schema.py

```diff
@@ -1451,6 +1451,8 @@ def _unsubstituted_typevar_schema(self, typevar: typing.TypeVar) -> core_schema.
         constraints = typevar.__constraints__
         not_set = object()
         default = getattr(typevar, '__default__', not_set)
+        if default is None:
+            default = not_set
```

Comment:
Can the default not be `None`? How does this differentiate `default=None` from no default?

Reply:
According to the PEP:
- when default is unset, `__default__` would be `None`
- if `default=None`, `__default__` should be `NoneType`

Reply:
Can we get rid of our internal `unset` thing then?

Reply:
Hmm, yes - let me try and see if that works

Reply:
Done :)